fil_fungible_token/src/token/mod.rs

```diff
@@ -45,40 +45,36 @@ type Result<T> = std::result::Result<T, TokenError>;
 /// Library functions that implement core FRC-??? standards
 ///
 /// Holds injectable services to access/interface with IPLD/FVM layer.
-pub struct Token<BS, MSG>
+pub struct Token<'st, BS, MSG>
 where
-    BS: IpldStore + Clone,
+    BS: IpldStore,
     MSG: Messaging,
 {
     /// Injected blockstore. The blockstore must reference the same underlying storage under Clone
     bs: BS,
     /// Minimal interface to call methods on other actors (i.e. receiver hooks)
     msg: MSG,
     /// In-memory cache of the state tree
-    state: TokenState,
+    state: &'st mut State,
 }
 
-impl<BS, MSG> Token<BS, MSG>
+impl<'st, BS, MSG> Token<'st, BS, MSG>
 where
-    BS: IpldStore + Clone,
+    BS: IpldStore,
     MSG: Messaging,
 {
     /// Creates a new token instance using the given blockstore and creates a new empty state tree
     ///
     /// Returns a Token handle that can be used to interact with the token state tree and the Cid
     /// of the state tree root
-    pub fn new(bs: BS, msg: MSG) -> Result<(Self, Cid)> {
-        let init_state = TokenState::new(&bs)?;
-        let cid = init_state.save(&bs)?;
-        let token = Self { bs, msg, state: init_state };
-        Ok((token, cid))
+    pub fn new(bs: BS, msg: MSG, state: &'st mut State) -> Result<Self> {
+        let token = Self { bs, msg, state };
+        Ok(token)
     }
 
-    /// For an already initialised state tree, loads the state tree from the blockstore and returns
-    /// a Token handle to interact with it
-    pub fn load(bs: BS, msg: MSG, state_cid: Cid) -> Result<Self> {
-        let state = TokenState::load(&bs, &state_cid)?;
-        Ok(Self { bs, msg, state })
+    /// Returns a mutable reference to underlying token-state
+    pub fn state(&mut self) -> &mut State {
```

Comment:
Thanks for following my quick hacks here. I think we can find better/safer flows. There are two main use cases I'm aware of right now, and they might each benefit from a helper.
1. Token state is the only state for an actor, or linked by CID. In this case we want to retain something like the old init_state path that creates state, and there's no need to expose it.
2. Token state is embedded in actor state. In this case we need to wrap, and this `state` method I added was to let the caller copy the state back out. But that's an avenue for error (forgetting to do that). If it's a mutable reference, maybe we don't need this at all? Another alternative is a `with(state, callback(token))` transaction-style method that exposes the Token API only within a closure, and ensures writeback.

Something like the callback method could work for (1) too.

Another alternative for (1) would be a `Rc` over the state, but we can probably avoid that.

I think it'll be very hard to get these APIs right without working an example actor of each type.